shared-kernel-infrastructure/src/main/java/org/dcsa/skernel/infrastructure/util/EnumUtil.java

```diff
@@ -0,0 +1,30 @@
+package org.dcsa.skernel.infrastructure.util;
+
+import lombok.experimental.UtilityClass;
+import org.dcsa.skernel.errors.exceptions.ConcreteRequestErrorMessageException;
+
+import java.util.Arrays;
+import java.util.List;
+
+@UtilityClass
+public class EnumUtil {
+  /**
+   * Splits a comma-separated string and converts each subpart into
+   * an enum.
+   */
+  public static <T extends Enum<T>> List<T> toEnumList(String values, Class<T> enumClass) {
```

Comment:
Wouldn't it be simplier to have `Class<T> enumClass` be replaced by `Function<String, T> valueOf` and then pass `ConcreteEnum::valueOf` as argument instead?

Reply:
As in, I think the stream would be simpler also in the form of:

```java
 return Arrays.stream(values.split(","))
      .map(String::trim)
      .map(valueOf)
      .onErrorSomething(e -> ConcreteRequestErrorMessageException.invalidInput(...)
      .toList();
```

Reply:
Simplified it using the builtin Enum.valueOf(type, str)